src/coreclr/hosts/coreconsole/coreconsole.cpp

```diff
@@ -596,14 +596,26 @@ void showHelp() {
 int __cdecl wmain(const int argc, const wchar_t* argv[])
 {
 	auto programPath = _wcsdup(argv[0]);
-	auto extension = wcsrchr(programPath, '.') + 1;
-	if (wcscmp(extension, L"exe") != 0) {
-		::wprintf(W("This executable needs to have 'exe' extension"));
-		return -1;
+	auto extension = wcsrchr(programPath, '.');
+	if (extension == NULL) {
+		// We were called without the extension so need a bigger buffer
+		size_t pathLen = wcslen(programPath);
+        size_t bufLen = pathLen + 5;
```

Comment:
nit - formatting is off - mix of tabs and spaces. Could you please convert tabs to spaces?